app/mailboxes/mailbox_helper.rb

```diff
@@ -24,15 +24,47 @@ def add_attachments_to_message
     return if @message.blank?
 
     processed_mail.attachments.last(Message::NUMBER_OF_PERMITTED_ATTACHMENTS).each do |mail_attachment|
-      attachment = @message.attachments.new(
-        account_id: @conversation.account_id,
-        file_type: 'file'
-      )
-      attachment.file.attach(mail_attachment[:blob])
+      if inline_attachment?(mail_attachment) && @mail.html_part.present?
+        upload_inline_image(mail_attachment)
+      elsif inline_attachment?(mail_attachment) && @mail.text_part.present?
+        embed_plain_text_email_with_inline_image(mail_attachment)
+      else
+        attachment = @message.attachments.new(
+          account_id: @conversation.account_id,
+          file_type: 'file'
+        )
+        attachment.file.attach(mail_attachment[:blob])
+      end
     end
     @message.save!
   end
 
+  def upload_inline_image(mail_attachment)
+    content_id = mail_attachment[:original].cid
+
+    @message.content_attributes[:email][:html_content][:full] = processed_mail.serialized_data[:html_content][:full].gsub(
+      "cid:#{content_id}", inline_image_url(mail_attachment[:blob]).to_s
+    )
+    @message.save!
+  end
+
+  def inline_attachment?(mail_attachment)
+    mail_attachment[:original].inline?
+  end
+
+  def embed_plain_text_email_with_inline_image(mail_attachment)
```

Comment:
Very rare case: if the user has no HTML content, not signature, no font changes, just plain text, and an image then we need to embed the images for the text_part of the email.